src/libcollections/string.rs

```diff
@@ -978,49 +978,6 @@ impl ops::Deref for String {
     }
 }
 
-/// Wrapper type providing a `&String` reference via `Deref`.
-#[unstable(feature = "collections")]
-#[deprecated(since = "1.2.0",
-             reason = "replaced with deref coercions or Borrow")]
-#[allow(deprecated)]
-pub struct DerefString<'a> {
```

Comment:
Note: I deleted this stuff since it's been deprecated for a while and porting it to the new stuff would be more effort than it's worth.